packages/beacon-node/test/unit/chain/opPools/aggregatedAttestationPool.test.ts

```diff
@@ -158,6 +160,127 @@ describe("AggregatedAttestationPool", () => {
+  }
+
+  (originalState as CachedBeaconStateElectra).previousEpochParticipation =
+    ssz.altair.EpochParticipation.toViewDU(epochParticipation);
+  (originalState as CachedBeaconStateElectra).currentEpochParticipation =
+    ssz.altair.EpochParticipation.toViewDU(epochParticipation);
+  originalState.commit();
+  let electraState: CachedBeaconStateAllForks;
+
+  let forkchoiceStub: MockedForkChoice;
+
+  beforeEach(() => {
+    pool = new AggregatedAttestationPool(config);
+    electraState = originalState.clone();
+    forkchoiceStub = getMockedForkChoice();
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("Multiple attestations with same attestation data different committee", () => {
+    // Attestation from committtee 0
+    const committeeBits0 = BitArray.fromSingleBit(MAX_COMMITTEES_PER_SLOT, 0);
+
+    const attestation0: Attestation<ForkPostElectra> = {
+      ...attestation,
+      aggregationBits: new BitArray(new Uint8Array(committeeLength / 8).fill(1), committeeLength),
+      committeeBits: committeeBits0,
+    };
+
+    // Attestation from committtee 1
+    const committeeBits1 = BitArray.fromSingleBit(MAX_COMMITTEES_PER_SLOT, 1);
+
+    const attestation1: Attestation<ForkPostElectra> = {
+      ...attestation,
+      aggregationBits: new BitArray(new Uint8Array(committeeLength / 8).fill(1), committeeLength),
+      committeeBits: committeeBits1,
+    };
+    // Attestation from committtee 2
+    const committeeBits2 = BitArray.fromSingleBit(MAX_COMMITTEES_PER_SLOT, 2);
+
+    const attestation2: Attestation<ForkPostElectra> = {
+      ...attestation,
+      aggregationBits: new BitArray(new Uint8Array(committeeLength / 8).fill(1), committeeLength),
+      committeeBits: committeeBits2,
+    };
+
+    // Attestation from committtee 3
+    const committeeBits3 = BitArray.fromSingleBit(MAX_COMMITTEES_PER_SLOT, 3);
+
+    const attestation3: Attestation<ForkPostElectra> = {
+      ...attestation,
+      aggregationBits: new BitArray(new Uint8Array(committeeLength / 8).fill(1), committeeLength),
+      committeeBits: committeeBits3,
+    };
+
+    pool.add(attestation0, attDataRootHex, attestation0.aggregationBits.getTrueBitIndexes().length, committees[0]);
+
+    pool.add(attestation1, attDataRootHex, attestation0.aggregationBits.getTrueBitIndexes().length, committees[1]);
+
+    pool.add(attestation2, attDataRootHex, attestation0.aggregationBits.getTrueBitIndexes().length, committees[2]);
+
+    pool.add(attestation3, attDataRootHex, attestation0.aggregationBits.getTrueBitIndexes().length, committees[3]);
+
+    forkchoiceStub.getBlockHex.mockReturnValue(generateProtoBlock());
+    forkchoiceStub.getDependentRoot.mockReturnValue(ZERO_HASH_HEX);
+
+    const blockAttestations = pool.getAttestationsForBlock(fork, forkchoiceStub, electraState);
+
+    expect(blockAttestations.length).toBe(1); // Expect attestations from committee 0, 1, 2 and 3 to be aggregated into one
```

Comment:
`blockAttestations.length` returns 4 before #7455 which is not correct. @twoeths